### PR TITLE
[MIG]website_mail : migrate to new api

### PR DESCRIPTION
**Pull request by bud-e**
Task : https://www.odoo.com/web#id=12029&view_type=form&model=project.task&menu_id=3942&action=327
Pad : https://pad.odoo.com/p/r.Yk2pVlLvxdw8HYUr

### CODE REVIEW

File: addons/website_mail/__openerp__.py [hunk 1]
```diff
@@ -25,7 +6,7 @@
     'summary': 'Website Module for Mail',
     'version': '0.1',
     'description': """Glue module holding mail improvements for website.""",
-    'author': 'OpenERP SA',
+    'author': 'Odoo SA',
```
Comment (vme-odoo) [inline addons/website_mail/__openerp__.py]: its Odoo S.A.

File: addons/website_mail/controllers/main.py [hunk 2]
```diff
@@ -1,99 +1,71 @@
...
         _message_is_follower = message_is_follower == 'on'
-        _object = request.registry[object]
+        _object = request.env[object]
 
         # search partner_id
         public_id = request.website.user_id.id
-        if uid != public_id:
-            partner_ids = [user_obj.browse(cr, uid, uid, context).partner_id.id]
+        if request.uid != public_id:
+            partner_ids = [User.browse(request.uid).partner_id.id]
```
Comment (vme-odoo) [inline addons/website_mail/controllers/main.py]: use self.env.user, no need to browse

File: addons/website_mail/controllers/main.py [hunk 3]
```diff
@@ -1,99 +1,71 @@
...
 
     @http.route(['/website_mail/follow'], type='json', auth="public", website=True)
     def website_message_subscribe(self, id=0, object=None, message_is_follower="on", email=False, **post):
-        cr, uid, context = request.cr, request.uid, request.context
-
-        partner_obj = request.registry['res.partner']
-        user_obj = request.registry['res.users']
-
+        Partner = request.env['res.partner']
+        User = request.env['res.users']
```
Comment (vme-odoo) [inline addons/website_mail/controllers/main.py]: remove this

File: addons/website_mail/controllers/main.py [hunk 4]
```diff
@@ -1,99 +1,71 @@
...
+        Obj = request.env[model]
 
         partner_id = None
         public_id = request.website.user_id.id
-        if uid != public_id:
-            partner_id = users_obj.browse(cr, SUPERUSER_ID, uid, context).partner_id
+        if request.uid != public_id:
+            partner_id = User.sudo().browse(request.uid).partner_id
```
Comment (vme-odoo) [inline addons/website_mail/controllers/main.py]: use self.env.user

File: addons/website_mail/models/mail_message.py [hunk 5]
```diff
@@ -1,89 +1,66 @@
...
+        for message in self:
             if message.subject:
-                res[message.id] = message.subject
+                detail = message.subject
             else:
                 plaintext_ct = '' if not message.body else html2plaintext(message.body)
-                res[message.id] = plaintext_ct[:30] + '%s' % (' [...]' if len(plaintext_ct) >= 30 else '')
-        return res
+                detail = plaintext_ct[:30] + '%s' % (' [...]' if len(plaintext_ct) >= 30 else '')
+        self.description = detail
```
Comment (vme-odoo) [inline addons/website_mail/models/mail_message.py]: no need of iteration here

File: addons/website_mail/models/mail_message.py [hunk 6]
```diff
@@ -1,89 +1,66 @@
...
+    @api.model
+    def _search(self, args, offset=0, limit=None, order=None,
+                count=False, access_rights_uid=None):
         """ Override that adds specific access rights of mail.message, to restrict
         messages to published messages for public users. """
-        if uid != SUPERUSER_ID:
-            group_ids = self.pool.get('res.users').browse(cr, uid, uid, context=context).groups_id
-            group_user_id = self.pool.get("ir.model.data").get_object_reference(cr, uid, 'base', 'group_public')[1]
+        if self.env.uid != self.sudo().env.uid:
+            group_ids = self.env['res.users'].browse(self.env.uid).groups_id
```
Comment (vme-odoo) [inline addons/website_mail/models/mail_message.py]: same as above

File: addons/website_mail/controllers/main.py (hunk 2 above)
Comment (sha-odoo) [inline addons/website_mail/controllers/main.py]: done.

File: addons/website_mail/controllers/main.py (hunk 3 above)
Comment (sha-odoo) [inline addons/website_mail/controllers/main.py]: ok, done.

File: addons/website_mail/controllers/main.py (hunk 4 above)
Comment (sha-odoo) [inline addons/website_mail/controllers/main.py]: done.

File: addons/website_mail/models/mail_message.py (hunk 5 above)
Comment (sha-odoo) [inline addons/website_mail/models/mail_message.py]: done.

File: addons/website_mail/controllers/main.py [hunk 7]
```diff
@@ -1,99 +1,68 @@
...
         _message_is_follower = message_is_follower == 'on'
-        _object = request.registry[object]
+        _object = request.env[object]
 
         # search partner_id
         public_id = request.website.user_id.id
-        if uid != public_id:
-            partner_ids = [user_obj.browse(cr, uid, uid, context).partner_id.id]
+        if request.uid != public_id:
+            partner_ids = [request.env.user.partner_id.id]
```
Comment (rga-odoo) [inline addons/website_mail/controllers/main.py]: partner_ids =  request.env.user.partner_id.ids

File: addons/website_mail/models/mail_message.py [hunk 8]
```diff
@@ -1,89 +1,64 @@
...
+                count=False, access_rights_uid=None):
         """ Override that adds specific access rights of mail.message, to restrict
         messages to published messages for public users. """
-        if uid != SUPERUSER_ID:
-            group_ids = self.pool.get('res.users').browse(cr, uid, uid, context=context).groups_id
-            group_user_id = self.pool.get("ir.model.data").get_object_reference(cr, uid, 'base', 'group_public')[1]
+        if self.env.uid != self.sudo().env.uid:
+            group_ids = self.env.user.groups_id
+            group_user_id = self.env["ir.model.data"].get_object_reference('base', 'group_public')[1]
             if group_user_id in [group.id for group in group_ids]:
```
Comment (rga-odoo) [inline addons/website_mail/models/mail_message.py]: `[group.id for group in group_ids]` is identical to `group_ids.ids`

File: addons/website_mail/controllers/main.py (hunk 7 above)
Comment (sha-odoo) [inline addons/website_mail/controllers/main.py]: ok. done

File: addons/website_mail/models/mail_message.py (hunk 8 above)
Comment (sha-odoo) [inline addons/website_mail/models/mail_message.py]: ok done.. thanks.. :+1:

File: addons/website_mail/controllers/main.py [hunk 9]
```diff
@@ -1,99 +1,68 @@
...
         else:
             # mail_thread method
-            partner_ids = _object._find_partner_from_emails(
-                cr, SUPERUSER_ID, _id, [email], context=context, check_followers=True)
+            partner_ids = _object.sudo()._find_partner_from_emails(_id, [email], check_followers=True)
             if not partner_ids or not partner_ids[0]:
-                partner_ids = [partner_obj.create(cr, SUPERUSER_ID, {'name': email, 'email': email}, context=context)]
-
+                partner_ids = [Partner.sudo().create({'name': email, 'email': email})]
```
Comment (rga-odoo) [inline addons/website_mail/controllers/main.py]: create return new recordset so partner_ids have recordset in list not id

File: addons/website_mail/models/mail_message.py [hunk 10]
```diff
@@ -1,89 +1,64 @@
...
             - read:
                 - raise if the type is comment and subtype NULL (internal note)
         """
-        if uid != SUPERUSER_ID:
-            group_ids = self.pool.get('res.users').browse(cr, uid, uid, context=context).groups_id
-            group_user_id = self.pool.get("ir.model.data").get_object_reference(cr, uid, 'base', 'group_public')[1]
-            if group_user_id in [group.id for group in group_ids]:
-                cr.execute('SELECT id FROM "%s" WHERE website_published IS FALSE AND id = ANY (%%s)' % (self._table), (ids,))
-                if cr.fetchall():
+        if self.env.uid != self.sudo().env.uid:
```
Comment (rga-odoo) [inline addons/website_mail/models/mail_message.py]: why not use SUPERUSER_ID ?

File: addons/website_mail/controllers/main.py (hunk 9 above)
Comment (sha-odoo) [inline addons/website_mail/controllers/main.py]: ok, done.

**Pull request closed without merging**